src/displayapp/screens/ScreenList.h

```diff
@@ -9,16 +9,33 @@
 namespace Pinetime {
   namespace Applications {
     namespace Screens {
+
+      enum class ScreenListModes {UpDown, RightLeft, LongPress};
       template <size_t N>
       class ScreenList : public Screen {
         public:
-          ScreenList(DisplayApp* app, std::array<std::function<std::unique_ptr<Screen>()>, N>&& screens)
-          : Screen(app), screens{std::move(screens)}, current{this->screens[0]()} {
 
+          ScreenList(            
+            DisplayApp* app, 
+            uint8_t initScreen,
+            std::array<std::function<std::unique_ptr<Screen>()>, N>&& screens, 
+            ScreenListModes mode            
+          )
+          : Screen(app), 
+            initScreen{initScreen},
+            screens{std::move(screens)}, 
+            mode{mode},
+            current{this->screens[initScreen]()}
+          {
+            screenIndex = initScreen;
           }
 
           ~ScreenList() override {
-
+            current.reset(nullptr);
```

Comment:
I think this is not necessary : the using pointer will be deleted by the default destructor, and this class does not allocate any LVGL objects that must be de-allocated.
Or did I miss something?